### PR TITLE
Babel plugin and React Native example

### PR DESCRIPTION
**Pull request by hrastnik**
I've managed to get the babel plugin working - it successfully replaces .telefunc files with an RPC call. Had to grab the exports from the Babel AST, as the WASM implementation is async, and Babel plugins don't work well with async.

I can't seem to get the telefunc server middleware to actually run the telefunc files and would appreciate help there.

One more thing I noticed: When installing the app via `pnpm android` or `pnpm ios` it installs the app to the device and runs the Metro bundler as usual, but for some reason Metro doesn't detect the babel config. It works fine when running the bundler via the `pnpm start` command.
~I suspect the reason is that the `pnpm android|ios` commands run the packager from the home folder instead of the project root.~
~It could also be an issue with the monorepo - React Native doesn't play well with monorepos. I've had to change some paths on the native side to get it to build.~
After some investigation I see that when Metro is launched automatically after app installation (through `pnpm android|ios`) its working directory is the workspace root instead of the React Native project root so it fails to detect the babel config. Note that this is problem only when the project structure is a monorepo and I believe this should be fixed by the RN team. A workaround to get the telefunc example working is to simply start the bundler via `pnpm start`

### CODE REVIEW

**Comment by brillout** [comment]
The example is now working ;-).

I'm going to replace the `telefuncConfig.telefuncFile` trick with a `fast-glob` crawler. Also I've a couple of review nitpicks coming :-).

File: examples/react-native/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "main": "node_modules/expo/AppEntry.js",
```
Comment (brillout) [inline examples/react-native/package.json:2]: Is this required? Can we remove it?

File: examples/react-native/app.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      "image": "./assets/splash.png",
+      "resizeMode": "contain",
+      "backgroundColor": "#ffffff"
+    },
+    "updates": {
+      "fallbackToCacheTimeout": 0
+    },
+    "assetBundlePatterns": [
+      "**/*"
+    ],
```
Comment (brillout) [inline examples/react-native/app.json:18]: Are these lines required? Can we remove them?

**Review by brillout — CHANGES_REQUESTED** [review]
I've made quite a lot of changes. It still works on web, but does it also still work on iOS/Android?

We're almost there :-).

**Comment by brillout** [comment]
Also, I did a small change to the babel plugin:  
https://github.com/vikejs/telefunc/pull/18/commits/ff0d78bf3ab481df724d0c6a47f495dc08cd4464#diff-478665d828e185775647a31e833dd64fd1eeed5137a5a0230be0fd4a23382896L18 &mdash; is it ok to remove `subNode.exportKind === 'value'`?

File: examples/react-native/package.json (hunk 1 above)
Comment (hrastnik) [reply examples/react-native/package.json:2]: I believe this is required for Metro to work. It's comes by default with the blank Expo project.

File: examples/react-native/app.json (hunk 2 above)
Comment (hrastnik) [reply examples/react-native/app.json:18]: These are also added by default in a blank Expo project. I tried removing them and the project seems to be working fine without them.

**Comment by hrastnik** [comment]
@brillout  As far as I can tell after some research, the `subNode.exportKind` can be set to `type` when exporting a type like `export type X = string;`. I think it shouldn't cause issues if you remove it.

**Comment by brillout** [comment]
👍 So we are ready to merge?

**Comment by hrastnik** [comment]
Yes, I believe it's all good to go!

**Comment by brillout** [comment]
Done :-).

Can't wait to hear your experience using Telefunc with React Native :-).

**Comment by brillout** [comment]
Changes released in `v0.1.17`.

**Pull request merged**